src/net/larse/lcms/algorithms/VCT.java

```diff
@@ -1085,31 +897,31 @@ private int getMaxForLength(double minForUd, double maxForUd) {
       }
 
       // Move the year pointer ahead
-      if (yearCount > 0) {
-        i += yearCount;
-      } else {
-        i++;
-      }
+      i += (yearCount > 0) ? yearCount : 1;
     }
 
     // Initialize the container to hold means and standard deviations of
     // the maximum forest streak
-    for (i = 0; i < N_BANDS; i++) {
-      this.meanForUdBx[i] = 25.4;
-      this.sdForUdBx[i] = 25.4;
-    }
+    Arrays.fill(this.meanForUdBx, 25.4);
+    Arrays.fill(this.sdForUdBx, 25.4);
 
     // If there is a forest streak, calculate statistics on this streak
+    // TODO: Mean and StandardDeviation are somehow giving very slightly
+    // different values than getSliceMean / getSliceStd to the point where
+    // pixels change lcType in further logic based on threshold values.
+    // What to do on this?
     if (maxLength > 0) {
-      Mean m = new Mean();
+      // Mean m = new Mean();
       for (int j = 0; j < N_BANDS; j++) {
-        this.meanForUdBx[j] = m.evaluate(this.ud[j], iStart, iEnd-iStart);
+        // this.meanForUdBx[j] = m.evaluate(this.ud[j], iStart, iEnd - iStart);
+        this.meanForUdBx[j] = getSliceMean(this.ud[j], iStart, iEnd);
```

Comment:
For reasons I can't quite figure out, we get very slightly different answers from using `Mean` and `StandardDeviation` vs `getSliceMean` and `getSliceStd`.  This difference is sometimes enough to affect the logic of a pixel's `lcType`.  I've opted for the latter in order to keep tests consistent, but it's probably better to derive a new set of test rasters and start using `Mean` and `StandardDeviation`?